src/main/java/com/theelixrlabs/healthcare/service/DoctorPatientAssignmentService.java

```diff
@@ -0,0 +1,97 @@
+import com.theelixrlabs.healthcare.dto.DoctorPatientAssignmentDTO;
+import com.theelixrlabs.healthcare.exceptionHandler.CustomException;
+import com.theelixrlabs.healthcare.model.DoctorPatientAssignmentModel;
+import com.theelixrlabs.healthcare.repository.DoctorRepository;
+import com.theelixrlabs.healthcare.repository.PatientRepository;
+import org.springframework.context.MessageSource;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Service
+public class DoctorPatientAssignmentService {
+
+    private final DoctorPatientAssignmentRepository doctorPatientAssignmentRepository;
+    private final PatientRepository patientRepository;
+    private final DoctorRepository doctorRepository;
+    private final MessageSource messageSource;
+
+    public DoctorPatientAssignmentService(DoctorPatientAssignmentRepository doctorPatientAssignmentRepository,
+                                          PatientRepository patientRepository, DoctorRepository doctorRepository,
+                                          MessageSource messageSource) {
+        this.doctorPatientAssignmentRepository = doctorPatientAssignmentRepository;
+        this.patientRepository = patientRepository;
+        this.doctorRepository = doctorRepository;
+        this.messageSource = messageSource;
+    }
+
+
+    private void validateDoctorPatientAssignmentDTO(DoctorPatientAssignmentDTO doctorPatientAssignmentDTO) throws CustomException {
+        UUID doctorId, patientId;
+        try {
+            doctorId = UUID.fromString(doctorPatientAssignmentDTO.getDoctorID());
+        } catch (IllegalArgumentException illegalArgumentException) {
+            throw new CustomException(DoctorPatientAssignmentConstants.DOCTOR_ID_INVALID_KEY, messageSource);
+        }
+        try {
+            patientId = UUID.fromString(doctorPatientAssignmentDTO.getPatientID());
+        } catch (IllegalArgumentException illegalArgumentException) {
+            throw new CustomException(DoctorPatientAssignmentConstants.PATIENT_ID_INVALID_KEY, messageSource);
+        }
+        if (!doctorRepository.existsById(doctorId))
+            throw new CustomException(DoctorPatientAssignmentConstants.DOCTOR_NOT_FOUND_KEY, messageSource);
+        if (!patientRepository.existsById(patientId))
+            throw new CustomException(DoctorPatientAssignmentConstants.PATIENT_NOT_FOUND_KEY, messageSource);
+    }
+
+    /**
+     * Checks if doctor is previously assigned to same patient and if not then saves the doctorPatientAssignmentDTO
+     *
+     * @param doctorPatientAssignmentDTO DTO object containing doctorId and patientId
+     * @return DoctorPatientAssignmentDTO
+     * @throws CustomException if any exception occurs
+     */
+    public DoctorPatientAssignmentDTO assignDoctorToPatient(DoctorPatientAssignmentDTO doctorPatientAssignmentDTO) throws CustomException {
```

Comment:
Where does assignment is happening ?
here you are only calling the repo to get List of assigned data and validating its un-assignment
Also this operation should be done with the implementation of Mongo many to many relationship between doctors and patients collection !

Reply:
Resolved